[🐛 Bug]: OneShotNode.java is not part of the chrome node

### What happened?

When I try to run a pod as a SingleShotNode - then the node throws a class not found exception. 
This node implementation helps to shutdown nodes after a session has been completed. So scaling the nodes is much easier.

This relates also to this issue: https://github.com/SeleniumHQ/selenium/issues/9845

### Command used to start Selenium Grid with Docker

```shell
apiVersion: v1
kind: Pod
metadata:
  name: chrome-node-2
spec:
  volumes:
    - name: dshm
      emptyDir:
        medium: Memory
  restartPolicy: Never
  containers:
    - name: selenium-node-chrome
      image: selenium/node-chrome:4
      ports:
        - containerPort: 5555
      volumeMounts:
        - mountPath: /dev/shm
          name: dshm
      env:
        - name: SE_EVENT_BUS_HOST
          value: "selenium-hub"
        - name: SE_EVENT_BUS_SUBSCRIBE_PORT
          value: "4443"
        - name: SE_EVENT_BUS_PUBLISH_PORT
          value: "4442"
        - name: NODE_IMPLEMENTATION
          value: "org.openqa.selenium.grid.node.k8s.OneShotNode"
      resources:
        limits:
          memory: "550Mi"
          cpu: "500m"
        requests:
          memory: "350Mi"
          cpu: "100m"
```


### Relevant log output

```shell
2022-02-24 08:39:25,554 INFO Included extra file "/etc/supervisor/conf.d/selenium.conf" during parsing
2022-02-24 08:39:25,555 INFO supervisord started with pid 8
2022-02-24 08:39:26,559 INFO spawned: 'xvfb' with pid 10
2022-02-24 08:39:26,563 INFO spawned: 'vnc' with pid 11
2022-02-24 08:39:26,567 INFO spawned: 'novnc' with pid 12
2022-02-24 08:39:26,572 INFO spawned: 'selenium-node' with pid 13
2022-02-24 08:39:26,857 INFO success: xvfb entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2022-02-24 08:39:26,858 INFO success: vnc entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2022-02-24 08:39:26,858 INFO success: novnc entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2022-02-24 08:39:26,858 INFO success: selenium-node entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
Generating Selenium Config
Configuring server...
Setting up SE_NODE_HOST...
Setting up SE_NODE_PORT...
Setting up SE_NODE_GRID_URL...
Selenium Grid Node configuration: 
[events]
publish = "tcp://selenium-hub:4442"
subscribe = "tcp://selenium-hub:4443"
[node]
session-timeout = "300"
override-max-sessions = false
detect-drivers = false
max-sessions = 1
[[node.driver-configuration]]
display-name = "chrome"
stereotype = '{"browserName": "chrome", "browserVersion": "98.0", "platformName": "Linux"}'
max-sessions = 1
Starting Selenium Grid Node...
08:39:32.062 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
08:39:32.157 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
08:39:33.258 INFO [UnboundZmqEventBus.<init>] - Connecting to tcp://selenium-hub:4442 and tcp://selenium-hub:4443
08:39:33.866 INFO [UnboundZmqEventBus.<init>] - Sockets created
08:39:34.868 INFO [UnboundZmqEventBus.<init>] - Event bus ready
08:39:35.850 INFO [NodeServer.createHandlers] - Reporting self as: http://10.233.64.30:5555
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.openqa.selenium.grid.Bootstrap.runMain(Bootstrap.java:77)
	at org.openqa.selenium.grid.Bootstrap.main(Bootstrap.java:70)
Caused by: org.openqa.selenium.grid.config.ConfigException: java.lang.ClassNotFoundException: org.openqa.selenium.grid.node.k8s.OneShotNode
	at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:115)
	at org.openqa.selenium.grid.node.config.NodeOptions.getNode(NodeOptions.java:148)
	at org.openqa.selenium.grid.node.httpd.NodeServer.createHandlers(NodeServer.java:127)
	at org.openqa.selenium.grid.node.httpd.NodeServer.asServer(NodeServer.java:183)
	at org.openqa.selenium.grid.node.httpd.NodeServer.execute(NodeServer.java:230)
	at org.openqa.selenium.grid.TemplateGridCommand.lambda$configure$4(TemplateGridCommand.java:129)
	at org.openqa.selenium.grid.Main.launch(Main.java:83)
	at org.openqa.selenium.grid.Main.go(Main.java:57)
	at org.openqa.selenium.grid.Main.main(Main.java:42)
	... 6 more
Caused by: java.lang.ClassNotFoundException: org.openqa.selenium.grid.node.k8s.OneShotNode
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at org.openqa.selenium.grid.config.ClassCreation.callCreateMethod(ClassCreation.java:35)
	at org.openqa.selenium.grid.config.MemoizedConfig.lambda$getClass$4(MemoizedConfig.java:100)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1737)
	at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:95)
	... 14 more
Exception in thread "Thread-0" java.lang.NullPointerException
	at org.openqa.selenium.grid.node.httpd.NodeServer.lambda$new$0(NodeServer.java:79)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-02-24 08:39:36,282 INFO exited: selenium-node (exit status 1; not expected)
```


### Operating System

Linux

### Docker Selenium version (tag)

selenium/node-chrome:4

--- a/NodeBase/Dockerfile.txt
+++ b/NodeBase/Dockerfile.txt
@@ -165,6 +165,9 @@ ENV START_NO_VNC true
 # Path to the Configfile
 ENV CONFIG_FILE=/opt/selenium/config.toml
 ENV GENERATE_CONFIG true
+# Drain the Node after N sessions. 
+# A value higher than zero enables the feature
+ENV DRAIN_AFTER_SESSION_COUNT 0
 
 #========================
 # Selenium Configuration

--- a/NodeBase/generate_config
+++ b/NodeBase/generate_config
@@ -46,6 +46,7 @@ fi
 echo "session-timeout = \"${SE_NODE_SESSION_TIMEOUT}\"" >> "$FILENAME"
 echo "override-max-sessions = ${SE_NODE_OVERRIDE_MAX_SESSIONS}" >> "$FILENAME"
 echo "detect-drivers = false" >> "$FILENAME"
+echo "drain-after-session-count = ${DRAIN_AFTER_SESSION_COUNT}" >> "$FILENAME"
 echo "max-sessions = ${SE_NODE_MAX_SESSIONS}
 " >> "$FILENAME"
 

--- a/README.md
+++ b/README.md
@@ -602,7 +602,8 @@ variables.
 Here is an example with the default values of these environment variables:
 
 ```bash
-$ docker run -d -e SE_EVENT_BUS_HOST=<event_bus_ip|event_bus_name> -e SE_EVENT_BUS_PUBLISH_PORT=4442 -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 --shm-size="2g" selenium/node-chrome:4.1.3-20220427
+$ docker run -d -e SE_EVENT_BUS_HOST=<event_bus_ip|event_bus_name> -e SE_EVENT_BUS_PUBLISH_PORT=4442 \ 
+  -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 --shm-size="2g" selenium/node-chrome:4.1.3-20220427
 ```
 
 ### Setting Screen Resolution
@@ -612,7 +613,7 @@ These settings can be adjusted by specifying `SCREEN_WIDTH`, `SCREEN_HEIGHT`, `S
 environmental variables when starting the container.
 
 ``` bash
-docker run -d -e SCREEN_WIDTH=1366 -e SCREEN_HEIGHT=768 -e SCREEN_DEPTH=24 -e SCREEN_DPI=74 selenium/standalone-firefox
+docker run -d -e SCREEN_WIDTH=1366 -e SCREEN_HEIGHT=768 -e SCREEN_DEPTH=24 -e SCREEN_DPI=74 selenium/standalone-firefox:4.1.3-20220427
 ```
 
 ### Grid Url and Session Timeout
@@ -661,10 +662,25 @@ To avoid starting the server you can set the `START_XVFB` environment variable t
 (or any other value than `true`), for example:
 
 ``` bash
-$ docker run -d --net grid -e SE_EVENT_BUS_HOST=selenium-hub -e SE_EVENT_BUS_PUBLISH_PORT=4442 -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 -e START_XVFB=false --shm-size="2g" selenium/node-chrome
+$ docker run -d --net grid -e SE_EVENT_BUS_HOST=selenium-hub -e SE_EVENT_BUS_PUBLISH_PORT=4442 \
+  -e SE_EVENT_BUS_SUBSCRIBE_PORT=4443 -e START_XVFB=false --shm-size="2g" selenium/node-chrome:4.1.3-20220427
 ```
 
 For more information, see this GitHub [issue](https://github.com/SeleniumHQ/docker-selenium/issues/567).
+
+### Stopping the Node/Standalone after N sessions have been executed
+
+In some environments, like Docker Swarm or Kubernetes, it is useful to shut down the Node or Standalone
+container after N tests have been executed. For example, this can be used in Kubernetes to terminate the
+pod and then scale a new one after N sessions. Set the environment variable `DRAIN_AFTER_SESSION_COUNT` to
+a value higher than zero to enable this behaviour. 
+
+``` bash
+$ docker run -e DRAIN_AFTER_SESSION_COUNT=5 --shm-size="2g" selenium/standalone-firefox:4.1.3-20220427
+```
+
+With the previous command, the Standalone container will shutdown after 5 sessions have been executed.
+
 ___
 
 ## Building the images

--- a/Standalone/generate_config
+++ b/Standalone/generate_config
@@ -26,6 +26,7 @@ fi
 echo "session-timeout = \"${SE_NODE_SESSION_TIMEOUT}\"" >> "$FILENAME"
 echo "override-max-sessions = ${SE_NODE_OVERRIDE_MAX_SESSIONS}" >> "$FILENAME"
 echo "detect-drivers = false" >> "$FILENAME"
+echo "drain-after-session-count = ${DRAIN_AFTER_SESSION_COUNT}" >> "$FILENAME"
 echo "max-sessions = ${SE_NODE_MAX_SESSIONS}
 " >> "$FILENAME"
 

